packages/atomic/src/components/commerce/product-template-components/atomic-product-image/atomic-product-image.tsx

```diff
@@ -0,0 +1,188 @@
+import {Product, ProductTemplatesHelpers} from '@coveo/headless/commerce';
+import {Component, h, Prop, Element, State, Method} from '@stencil/core';
+import {Bindings} from '../../../../components';
+import {
+  InitializeBindings,
+  InitializableComponent,
+} from '../../../../utils/initialization-utils';
+import {filterProtocol} from '../../../../utils/xss-utils';
+import {ImageCarousel} from '../../../common/image-carousel/image-carousel';
+import {ProductContext} from '../product-template-decorators';
+
+type Image = {
+  src: string;
+  alt: string;
+};
+
+/**
+ * The `atomic-product-image` component renders an image from a product field.
+ * @internal
+ */
+@Component({
+  tag: 'atomic-product-image',
+  styleUrl: 'atomic-product-image.pcss',
+  shadow: true,
+})
+export class AtomicProductImage implements InitializableComponent {
+  @InitializeBindings() public bindings!: Bindings;
+  @ProductContext() private product!: Product;
+  @Element() private host!: HTMLElement;
+  @State() private useFallback = false;
+  @State() private currentImage = 0;
+  @State() private images: Image[] = [];
+
+  public error!: Error;
+
+  /**
+   * The product field which the component should use. This will look for the field in the product object first, then in the product.additionalFields object.
+   */
+  @Prop({reflect: true}) field: string = 'ec_thumbnails';
+
+  /**
+   * An optional fallback image URL that will be used in case the specified image is not available or an error is encountered.
+   */
+  @Prop({reflect: true}) fallback?: string;
+
+  /**
+   * Moves to the previous image, when the carousel is activated.
+   */
+  @Method() public async previousImage() {
+    this.currentImage =
+      this.currentImage - 1 < 0
+        ? this.numberOfImages - 1
+        : this.currentImage - 1;
+  }
+
+  /**
+   * Moves to the next image, when the carousel is activated.
+   */
+  @Method() public async nextImage() {
+    this.currentImage = (this.currentImage + 1) % this.numberOfImages;
+  }
+
+  /**
+   * Navigates to the specified image index.
+   *
+   * @param index - The index of the image to navigate to.
+   */
+  @Method() public async navigateToImage(index: number) {
+    this.currentImage = index;
+  }
+
+  private logWarning(message: string) {
+    this.bindings.engine.logger.warn(message, this.host);
+  }
+
+  private handleImageError(event: Event) {
+    const image = event.target as HTMLImageElement;
+    const message = `The image url "${image.src}" is not valid or could not be loaded. You might want to add a "fallback" property.`;
+    if (this.fallback && image.src !== this.fallback) {
+      this.useFallback = true;
+      image.src = this.fallback;
+    } else if (image.src === this.fallback) {
+      this.logWarning('The fallback image failed to load.');
+    } else {
+      this.logWarning(message);
+    }
+  }
```

Comment:
Will fallback to provided fallback image, and log a warning if the fallback itself returns an error.